### PR TITLE
Rewrite withdraw view with real data

### PR DESCRIPTION
**Pull request by ptare**
Far from the ideal implementation. Because the original Withdraw component wasn't parameterized based on any incoming data, I pretty much had to scrap and rewrite it.

Some of this code could definitely be broken up/reused, but I want to write the Deposit component first, to get a better handle on the shared vs. unique logic.

Fix some percentage related bugs in ManagePositions.
Also fix a bug in ManagePositions where token value was calculated based
on the total supply, not the user's balance.

Issue #604.

Signed-off-by: Prasad Tare <prasad.tare@gmail.com>

### CODE REVIEW

File: sponsor-dapp-v2/src/views/ManagePositions.js [hunk 1]
```diff
@@ -246,12 +249,14 @@ function ManagePositions(props) {
 
                           <td>
                             <strong>
-                              {format(data.totalHoldings)} DAI ({data.collateralizationRatio}%)
+                              {format(data.totalHoldings)} DAI ({data.collateralizationRatio.toString()}%)
```
Comment (mrice32) [inline sponsor-dapp-v2/src/views/ManagePositions.js]: (Just for my own curiosity) Does the `{}` syntax not stringify it for you? What was it showing previously?

File: sponsor-dapp-v2/src/views/Withdraw.js [hunk 2]
```diff
@@ -1,68 +1,88 @@
...
-    this.state = {
-      allowedToProceed: false,
-      withdrawAmount: "",
-      newAmount: "115 %",
-      isLoading: false
-    };
-  }
+  const { drizzle, useCacheCall, useCacheSend } = drizzleReactHooks.useDrizzle();
+  const { web3 } = drizzle;
+  const { toBN, fromWei, toWei } = web3.utils;
```
Comment (mrice32) [inline sponsor-dapp-v2/src/views/Withdraw.js]: Good idea, this makes things way easier to read.

File: sponsor-dapp-v2/src/views/Withdraw.js [hunk 3]
```diff
@@ -73,8 +93,10 @@ class Withdraw extends Component {
 
               <div className="popup__head-entry">
                 <p>
-                  <strong>Your facility has a 110% collateralization requirement.</strong> You can withdraw collateral
-                  from your facility as long as you maintain this requirement.{" "}
+                  <strong>
+                    Your facility has a {fromWei(collateralizationRequirement)}% collateralization requirement.
+                  </strong>{" "}
```
Comment (mrice32) [inline sponsor-dapp-v2/src/views/Withdraw.js]: Just for my understanding, what does this `{" "}` element do?

Comment (ptare) [reply sponsor-dapp-v2/src/views/Withdraw.js]: I _think_ it just puts a space haha. That's what prettier puts in when something in your JSX is too long of a line.

File: sponsor-dapp-v2/src/views/ManagePositions.js (hunk 1 above)
Comment (ptare) [reply sponsor-dapp-v2/src/views/ManagePositions.js]: The BN wasn't getting displayed properly. Though this isn't fully correct either: we probably need to format the percents better/more consistently.

File: sponsor-dapp-v2/src/views/Withdraw.js (hunk 3 above)
Comment (mrice32) [reply sponsor-dapp-v2/src/views/Withdraw.js]: ooooooh, this makes sense. I've seen this in yours and Roz's code, and I had no idea where y'all were getting that syntax - thanks.

**Pull request merged**